source/mir/random/tinflex/internal/area.d

```diff
@@ -0,0 +1,650 @@
+module mir.random.tinflex.internal.area;
+
+import mir.random.tinflex.internal.types : Interval;
+import std.traits : ReturnType;
+
+/**
+Determines the hat and squeeze function of an interval.
+Based on Theorem 1
+*/
+void determineSqueezeAndHat(S)(ref Interval!S iv)
+in
+{
+    assert(iv.lx < iv.rx, "invalid interval");
+}
+body
+{
+    import mir.random.tinflex.internal.linearfun : emptyFun, secant, tangent;
+    import mir.random.tinflex.internal.types : determineType, FunType;
+
+    enum sec = "secant(iv.lx, iv.rx, iv.ltx, iv.rtx)";
+    enum t_l = "tangent(iv.lx, iv.ltx, iv.lt1x)";
+    enum t_r = "tangent(iv.rx, iv.rtx, iv.rt1x)";
+
+    // could potentially be saved for subsequent calls
+    FunType type = determineType(iv);
+    with(FunType) with(iv)
```

Comment:
> Minor: I would drop the `with(iv)` here; 

Hmm to be honest I am a huge fan of this style as it removes the repetitiveness from the `switch`. @9il what do you think?